nametoaddr.c

```diff
@@ -102,6 +102,10 @@ struct rtentry;		/* declarations in <net/if.h> */
 #include <pcap/namedb.h>
 #include "nametoaddr.h"
 
+#if defined(_MSC_VER) && !defined(strdup)
```

Comment:
`portability.h` should take care of this.  Why is this necessary?

Reply:
It should, but the change in commit 9268e8661ffa4fe25bb742699127c711cb6e9e23 leaves a function undefined warning in debug mode (ie. when _DEBUG is defined), which this change corrects. Without it:
`C:\Users\blazeable\GitHub\libpcap\nametoaddr.c(265): warning C4013: 'strdup' undefined; assuming extern returning int`

Reply:
So that seems to imply that either:

1. `crtdbg.h` is only being included in some of the files that include `portability.h` and use `strdup()`, so, with `_DEBUG` defined, some files aren't getting `strdup` defined by `crtdbg.h`;

1. some versions of MSVC don't have the definition of `strdup(s)` as a call to `_strdup_dbg()`.

In the first case, presumably they should be including `crtdbg.h`, so that the debugging version of `strdup()` is used everywhere.

In the second case, presumably we should do something such as

    #ifndef strdup
        #define strdup  _strdup
    #endif

inside `#ifdef _MSC_VER`.  @gvanem, what do you think here?

Reply:
I've changed `portability.h` to define `strdup` on MSVC iff it's not already defined, regardless of whether `_DEBUG` is defined or not.  That way, *if* `crtdbg.h` has been included before `portability.h`, and *if* it's already defined `strdup`, it won't redefine it, otherwise it *will* define it.

This requires that `crtdbg.h` be included before `portability.h`; I've checked in another change to move the includes of `sockutils.h`, which may include `crtdbg.h`, before includes of `portability.h` or of files that include `portability.h`.

On my Windows 7 VM, with VS 2015, that appears to eliminate build issues with `strdup`, both with Debug, Release, and RelWithDebugInfo builds.

@gvanem, is that an OK fix?  If not, what's better?

This should remove the need to define `strdup` in any `.c` file, or in any `.h` file other than `portability.h`.

Reply:
With the recent fixes, I get no warnings with `_DEBUG`.

But `<crtdbg.h>` included in `sockutils.h` should perhaps be moved to `portability.h`?  Besides, `pcap-stdinc.h` has now become so small, it could be moved into `pcap-types.h`.

Reply:
I've moved the include of `winsock2.h` into `pcap-types.h`, as it's needed to define `u_int`; `pcap-types.h` already includes `sys/types.h` to define `u_int` on UN\*X and MS-DOS.  (So does `pcap/pcap.h`, so perhaps `pcap-types.h` neither needs to include `winsock2.h` nor `sys/types.h`.)

I also moved the necessary other includes to the individual files, as needed; not all files need all of those includes.  That's also similar to what we do on UN\*X.

That renders `pcap-stdinc.h` unnecessary, so I got rid of it.